util/file_reader_writer.cc

```diff
@@ -219,6 +219,16 @@ Status WritableFileWriter::Append(const Slice& data) {
   return s;
 }
 
+Status WritableFileWriter::Pad(const size_t pad_bytes) {
+  assert(pad_bytes < kDefaultPageSize);
+  assert((buf_.Capacity() - buf_.CurrentSize()) >= pad_bytes);
+
+  pending_sync_ = true;
```

Comment:
is there a problem if the max buffer size is not a power of two?

Reply:
Good catch. I'll fix this to do a flush if there's not enough space in buf_.